src/main/java/akshan/Akshan.java

```diff
@@ -0,0 +1,274 @@
+
+    static {
+        COMMAND_PARSERS.put(CommandType.TODO, Akshan::parseTodo);
+        COMMAND_PARSERS.put(CommandType.DEADLINE, Akshan::parseDeadline);
+        COMMAND_PARSERS.put(CommandType.EVENT, Akshan::parseEvent);
+    }
+
+
+    /**
+     * Prints a horizontal line.
+     */
+    private static void printLine(){
+        System.out.println("_______________________________________________________________");
+    }
+
+    /**
+     * Main method to run the Akshan bot.
+     */
+    public static void main(String[] args) throws IOException {
+        Scanner input = new Scanner(System.in);
+        TaskList taskList = new TaskList();
+        StorageHandler storageHandler = new StorageHandler(taskList);
+
+        init();
+        String line = input.nextLine();
+
+        while (!line.equals(CommandType.BYE.getCommand())) {
+            try {
+                processCommand(line, taskList);
+            } catch (IllegalArgumentException e) {
+                System.out.println("Error: " + e.getMessage());
+            } catch (Exception e) {
+                System.out.println("An unexpected error occurred: " + e.getMessage());
+            }
+            printLine();
+            storageHandler.saveData(taskList);
+            line = input.nextLine();
+        }
+        bye();
+    }
+
+    /**
+     * Initializes the bot, prints the logo and a welcome message.
+     */
+    private static void init(){
+        System.out.println(LOGO);
+        printLine();
+        System.out.println("Hello! I'm " + NAME);
+        System.out.println("What can I do for you?");
+        printLine();
+    }
+
+    /**
+     * Exit bot, prints goodbye sequence.
+     */
+    private static void bye(){
+        System.out.println( "Bye. Hope to see you again soon!");
+        printLine();
+    }
+
+    /**
+     * Processes the user's command.
+     *
+     * @param command The user's input command.
+     * @param taskList The list of tasks.
+     * @throws IllegalArgumentException If the command is unknown or invalid.
+     */
+    private static void processCommand(String command, TaskList taskList) throws IllegalArgumentException {
+        String[] splitInput = command.split("/");
+        String commandTypeString = splitInput[0].split(" ")[0];
+        CommandType commandType;
+        try {
+            commandType = CommandType.fromString(commandTypeString);
+        } catch (IllegalArgumentException e) {
+            throw new IllegalArgumentException("Unknown command: " + commandTypeString);
+        }
+
+        switch (commandType) {
+        case LIST:
+            taskList.printList();
+            break;
+        case MARK:
+            processMarkUnmark(splitInput[0], taskList, true);
+            break;
+        case UNMARK:
+            processMarkUnmark(splitInput[0], taskList, false);
+            break;
+        case TODO:
+        case DEADLINE:
+        case EVENT:
```

Comment:
Maybe it would be good to explicitly mention that it is a fallthrough as mentioned [here](https://se-education.org/guides/conventions/java/basic.html#layout)

Reply:
Good point, will amend accordingly.